azblob/sas_service.go

```diff
@@ -24,6 +25,83 @@ type BlobSASSignatureValues struct {
 	ContentEncoding    string // rsce
 	ContentLanguage    string // rscl
 	ContentType        string // rsct
+	UserDelegationKey  UserDelegationKey
+}
+
+func (v BlobSASSignatureValues) NewIdentitySASQueryParameters(AccountName string) (SASQueryParameters, error) {
```

Comment:
Shouldn't the UDK be taken in as a parameter? Similar to how the sharedKeyCredential is passed into NewSASQueryParameters. Basically, the credential is taken in to sign the SAS.